packages/web-components/src/text-input/text-input.ts

```diff
@@ -0,0 +1,31 @@
+import { attr } from '@microsoft/fast-element';
+import { FASTTextField } from '@microsoft/fast-foundation';
+import { TextInputAppearance, TextInputControlSize } from './text-input.options.js';
+
+/**
+ * The base class used for constructing a fluent-text-input custom element
+ * @public
+ */
+export class TextInput extends FASTTextField {
+  /**
+   * Defines TextInput control size
+   *
+   * @public
+   * @default 'medium'
+   * @remarks
+   * HTML Attribute: control-size
+   */
+  @attr({ attribute: 'control-size' })
+  public controlSize: TextInputControlSize = TextInputControlSize.medium;
```

Comment:
I think the agreement is not to assign default value to the attributes unless completely necessary to reduce DOM size. Styles should be authored to honor the default values without explicitly setting them in javascript.